src/frameworks/flask/index.ts

```diff
@@ -0,0 +1,79 @@
+import { copy, mkdirp } from "fs-extra";
+import { mkdir, readFile, readdir, writeFile } from "fs/promises";
+import { join, relative } from "path";
+import { BuildResult, FrameworkType, SupportLevel } from "../interfaces";
+import { dirExistsSync } from "../../fsutils";
+import { findPythonCLI, hasPipDependency, spawnPython } from "../utils";
+import { spawnSync } from "child_process";
+
+export const name = "Flask";
+export const support = SupportLevel.Experimental;
+export const type = FrameworkType.Framework;
+
+export async function discover(cwd: string) {
+  if (!hasPipDependency("Flask", { cwd })) return;
+  const results = await getDiscoveryResults(cwd).catch(() => undefined);
+  if (!results) return;
+  const publicDirectory = relative(cwd, results.staticFolder);
+  return { mayWantBackend: true, publicDirectory };
+}
+
+export async function init(setup: any, config: any) {
+  const cwd = join(config.projectDir, setup.hosting.source);
+  await mkdirp(cwd);
+  const cli = findPythonCLI();
+  spawnSync(cli, ["-m", "venv", "venv"], { stdio: "ignore", cwd });
+  writeFile(join(cwd, "requirements.txt"), "Flask");
+  await spawnPython("pip", ["install", "-r", "requirements.txt"], cwd);
+  await writeFile(
+    join(cwd, "main.py"),
+    `from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello_world():
+    return "<p>Hello, World!</p>"
+`
+  );
+}
+
+export function build(): Promise<BuildResult> {
+  return Promise.resolve({ wantsBackend: true });
+}
+
+export async function ɵcodegenPublicDirectory(root: string, dest: string) {
+  const { staticFolder, staticUrlPath } = await getDiscoveryResults(root);
+  const staticDest = join(dest, staticUrlPath);
+  await mkdirp(staticDest);
+  if (dirExistsSync(staticFolder)) {
+    await copy(staticFolder, staticDest);
+  }
+}
+
+export async function ɵcodegenFunctionsDirectory(root: string, dest: string) {
+  await mkdir(join(dest, "src"), { recursive: true });
+  // COPY everything except venv
+  const files = await readdir(root);
```

Comment:
maybe this should be `directories` instead?
```suggestion
  const directories = await readdir(root);
```

Reply:
Well, there can be files and directories 😅